src/FilterPlugin.php

```diff
@@ -22,14 +22,20 @@ class FilterPlugin implements Swift_Events_SendListener
 	 */
 	private $blacklist;
 
+    /**
+     * @var null|ExtraChecksInterface
+     */
+	private $extraChecks;
+
 	/**
 	 * @param array $whitelist
 	 * @param array $blacklist
 	 */
-	public function __construct(array $whitelist = [], array $blacklist = [])
+	public function __construct(array $whitelist = [], array $blacklist = [], ?ExtraChecksInterface $extraChecks = null)
```

Comment:
In order to design [the above suggestion]() in the API of the plugin we need to decide whether to introduce a breaking change in the API for this feature (I think this is reasonable as whoever needs the feature could upgrade) or we could keep backwards compatibility by still allowing the whitelist/blacklist arrays as the first two arguments in the constructor and pass them on to the whitelist/blacklist filters internally.

I personally prefer to design the good API and release as 0.3.0.

The constructor could look like that:

```php

/** @var ExtraChecksInterface[] $filters
private filters = [];

/** @param ExtraChecksInterface[] $filters */
public function __construct(array $filters = [])
{
	$this->filters = $filters;
}
```

Implementation of the filtering would look closer to just:

```php
/**
 * @param  string $email
 * @return boolean
 */
public function filterEmail($email)
{
	foreach ($this->filters as $filter) {
		if (!$filter->check($email)) {
			return false;
		}
	}

	return true;
}
```

Here's an example usage in userland:
```php
$dispatcher->bindEventListener(new FilterPlugin([
	new WhitelistFilter(['example.com', 'foo@example.org']),
	new BlacklistFilter(['bar@example.com'])
]));
```